### ISSUE TITLE
Calling a saga that returns a value doesn't work

### ISSUE THREAD

**Issue by publicJorn** [jfairbank/redux-saga-test-plan#91]
Hi @jfairbank, I found another issue, given the same example code as in #85.
I think it's again related to call, although at a different level.

Saga 1 should be able to `yield call` saga 2. Saga 2 should be able to return a value that saga 1 can then use. [Here is a working example](https://codepen.io/publicJorn/pen/QpReGK?editors=0010)
Scroll down to `//// sagas`

I forked this repo and added these tests to [put.test.js](https://github.com/jfairbank/redux-saga-test-plan/blob/master/__tests__/expectSaga/assertions/put.test.js). You can copy-paste them if  you please:

```javascript
// update first import to:
import { put, call } from 'redux-saga/effects';

function actionFn() {
	return { type: 'CALLED_FROM_FUNCTION', payload: 42 }
}

function* putSagaFromFn() {
	const action = yield call(actionFn)
	console.log('ACTION EXISTS\n', action)
	yield put(action)
}

test('put assertion handles yielded value from function', () => {
	expectSaga(putSagaFromFn)
		.put({ type: 'CALLED_FROM_FUNCTION', payload: 42 })
		.run()
})


function* actionSaga() {
	// whatever effect to get payload
	return { type: 'CALLED_FROM_SAGA', payload: 42 }
}

function* putSagaFromSaga() {
	const action = yield call(actionSaga)
	console.log('ACTION UNDEFINED\n', action)
	yield put(action)
}

test('put assertion handles yielded value from other saga', () => {
	expectSaga(putSagaFromSaga)
		.put({ type: 'CALLED_FROM_SAGA', payload: 42 })
		.run()
})
```

The second test fails.

**Comment by jfairbank** [comment]
Thanks for catching this as well @publicJorn!

I was afraid that some bugs might pop up when I introduced providers to `expectSaga`. I have a fix in place that I need to finish testing, so I'll have a release out later today.

**Comment by jfairbank** [comment]
Fix is out in v2.3.6. Let me know if you have any issues. Thanks!

**Comment by publicJorn** [comment]
Works like a charm. Thanks!
